flang/lib/Lower/OpenACC.cpp

```diff
@@ -1343,11 +1345,21 @@ static void genWaitClause(Fortran::lower::AbstractConverter &converter,
   }
 }
 
+mlir::Type getTypeFromIvTypeSize(fir::FirOpBuilder &builder,
+                                 const Fortran::semantics::Symbol &ivSym) {
+  std::size_t ivTypeSize = ivSym.size();
+  if (ivTypeSize == 0)
+    llvm::report_fatal_error("unexpected induction variable size");
+  return builder.getIntegerType(ivTypeSize * 8);
```

Comment:
A comment explaining the mapping here would be useful. It is not immediately obvious what ivSym.size() produces or what 8 is.

Reply:
Added comment